src/custom-apps/components/WebhookHeaders/WebhookHeaders.tsx

```diff
@@ -0,0 +1,304 @@
+import { Button } from "@dashboard/components/Button";
+import CardTitle from "@dashboard/components/CardTitle";
+import Skeleton from "@dashboard/components/Skeleton";
+import TableRowLink from "@dashboard/components/TableRowLink";
+import { FormChange } from "@dashboard/hooks/useForm";
+import { removeAtIndex, updateAtIndex } from "@dashboard/utils/lists";
+import {
+  Card,
+  CardActions,
+  CardContent,
+  Table,
+  TableBody,
+  TableCell,
+  TableHead,
+  TextField,
+  Typography,
+} from "@material-ui/core";
+import { DeleteIcon, ExpandIcon, IconButton } from "@saleor/macaw-ui";
+import clsx from "clsx";
+import React, {
+  ChangeEvent,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from "react";
+import { FormattedMessage, useIntl } from "react-intl";
+
+import { WebhookFormData } from "../WebhookDetailsPage";
+import useStyles from "./styles";
+import { mapHeaders, stringifyHeaders } from "./utils";
+
+export interface WebhookHeadersProps {
+  data: WebhookFormData;
+  onChange: FormChange;
+}
+
+export interface Header {
+  name: string;
+  value: string;
+  error?: boolean;
+}
+
+const nameSeparator = ":";
+const nameInputPrefix = "name";
+const valueInputPrefix = "value";
+
+const WebhookHeaders: React.FC<WebhookHeadersProps> = ({
+  data: { customHeaders },
+  onChange,
+}) => {
+  const intl = useIntl();
+  const loaded = useRef(false);
+  const [expanded, setExpanded] = useState(false);
+  const classes = useStyles();
+  const headers = useMemo(() => mapHeaders(customHeaders), [customHeaders]);
+
+  const change = (event: ChangeEvent) => {
+    const { name, value } = event.target as HTMLTextAreaElement;
```

Comment:
```suggestion
  const onChange = ({ name, value } : ChangeEvent<HTMLTextAreaElement>) => {
```

Reply:
Chaged to `const change = ({ target }: ChangeEvent<HTMLTextAreaElement>) => {`